File: JS Exams/01.  Basics Exam Collection/Basics Online Exam - 18 and 19 July 2020/05. Best Player/05. Best Player.js
function bestPlayer(arg) {
  let name = goals = null
  for (i = 0; i < arg.length; i+= 2) 
    if (goals < +arg[i + 1])
      [name, goals] = [arg[i], +arg[i + 1]] 
  console.log(name, 'is the best player!', `\nHe has scored ${goals} ${goals < 3 ? 'goals.' : 'goals and made a hat-trick !!!'}`)
}

//-----------------------------------------------(2)------------------------------------------

function bestPlayer(arr) {
    const playersObj = {}

    while (true) {
        let name = arr.shift()
        if (name == 'END') break
        let goals = arr.shift()

        playersObj[name] = goals
        if (playersObj[name] >= 10) break

    }

    let sortByGoals = Object.entries(playersObj).sort((a, b) => b[1] - a[1])
    let best_player = sortByGoals[0][0]
    let goals = playersObj[best_player]

    console.log(`${best_player} is the best player!`)
    if (goals < 3) console.log(`He has scored ${goals} goals.`)
    else console.log(`He has scored ${goals} goals and made a hat-trick !!!`)
}

// bestPlayer((["Neymar", "2", "Ronaldo", "1", "Messi", "3", "END"]))

File: Python Exams/01. Basic Exam Collection/Basics Online Exam - 18 and 19 July 2020/05. Best Player/05. Best Player.py
players_obj = {}

while True:
    name = input()
    if name == 'END': break
    goals = int(input())
    players_obj[name] = goals

    if players_obj[name] >= 10:
        break

sort_by_goals = dict(sorted(players_obj.items(), key=lambda x: -x[1]))
best_player = list(sort_by_goals.keys())[0]
goals = players_obj[best_player]

print(f'{best_player} is the best player!')
if goals < 3:
    print(f'He has scored {goals} goals.')
else:
    print(f'He has scored {goals} goals and made a hat-trick !!!')


'''
Neymar
2
Ronaldo
1
Messi
3
END
'''